server/src/utils.ts

```diff
@@ -186,7 +187,7 @@ async function validateSamePageLinks(xmlData: Document): Promise<Diagnostic[]> {
       LINK_DIAGNOSTIC_SOURCE
     )
     const linkTargetId: string = linkElement.getAttribute('target-id')
-    if (linkTargetId === '') {
+    if ((linkTargetId === '') || (validTargetIds.has(linkTargetId))) {
```

Comment:
What happens if the user deletes a target, but the link is preserved? Will the cache be stale and cause an invalid lack of an error?

Reply:
No, because this cache is intra-event and not maintained across events (for some of the reasons mentioned in comments in the parent issue, I think trying to do the latter will lead to various state sync issues). So if you edit / delete a target, errors will show for all links that reference it.

Reply:
Got it. So am I correct that we will only receive a hit in the cache when the same element is targeted by more than one link?

Reply:
Yes, that is correct.

Reply:
Neat. Did this help performance with the modules that took a while to process events?

Reply:
Yeah -- with both changes combined, anecdotally, a page I was using to test (`m58280` from university physics) went from ~1000ms to ~500ms with the combination of both of these (it had 61 links of which 56 were intra-page and 5 other page with IDs).

Reply:
Or, even better, I can just pull the xpath query [out of the loop altogether](https://github.com/openstax/poet/pull/20/commits/3b011dd2e8bc268bf403a83de128c65e56efefd4) for same page links and build up the valid / duplicate IDs before iterating through the links. Now that same page is down to ~200ms. Should have done that the first time...sigh...